parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -517,7 +523,67 @@ public PageReadStore readNextRowGroup() throws IOException {
     return columnChunkPageReadStore;
   }
 
+  /**
+   * Reads the requested predicate columns dictionaries from the row group at the current file position.
+   * Not reading all the requested columns, only read predicate columns.
+   * eg. select columnA, columnB, columnC from table where columnC > 10
+   *     only reads columnC's dictionary
+   * @param requestedSchema the predicate columns to read
+   * @throws IOException if an error occurs while reading
+   * @return the DictionaryPageReadStore which can provide DictionaryPageReaders for each column.
+   */
+  public DictionaryPageReadStore readNextRowGroupDictionaries(MessageType requestedSchema) throws IOException {
+    if (currentBlock == blocks.size()) {
+      return null;
+    }
+    BlockMetaData block = blocks.get(currentBlock);
+    if (block.getRowCount() == 0) {
+      throw new RuntimeException("Illegal row group of 0 rows");
+    }
+    ColumnChunkDictionaryPageReadStore dictionaryPageReadStore = new ColumnChunkDictionaryPageReadStore();
+    for (ColumnChunkMetaData columnChunkMetaData : block.getColumns()) {
+      for (ColumnDescriptor columnDescriptor : requestedSchema.getColumns()) {
+        if (columnChunkMetaData.getPath().equals(ColumnPath.get(columnDescriptor.getPath())) &&
+            isOnlyDictionaryEncodingPages(columnChunkMetaData.getEncodings())) {
+          long startingPosition = columnChunkMetaData.getStartingPos();
+          this.f.seek(startingPosition);
+
+          int totalSize = (int) columnChunkMetaData.getTotalSize();
+          byte[] buffer = new byte[totalSize];
+          this.f.readFully(buffer);
+
+          DictionaryStream dictionaryStream = new DictionaryStream(buffer, 0);
+          DictionaryPage compressedPage = dictionaryStream.readDictionaryPage();
+          if (compressedPage == null) {
+            dictionaryPageReadStore.addColumn(columnDescriptor, null);
+            break;
+          }
 
+          BytesDecompressor decompressor = codecFactory.getDecompressor(columnChunkMetaData.getCodec());
+          ColumnChunkDictionaryPageReader dictionaryPageReader = new ColumnChunkDictionaryPageReader(decompressor, compressedPage);
+          dictionaryPageReadStore.addColumn(columnDescriptor, dictionaryPageReader);
+        }
+      }
+    }
+    return dictionaryPageReadStore;
+  }
+
+  // are all the value pages dictionary encoded
+  private boolean isOnlyDictionaryEncodingPages(Set<Encoding> encodings) {
+    // if values have more than one encodings, definitely not dictionary encoding only
+    if (encodings.size() > 3) {
+      return false;
+    }
```

Comment:
@julienledem I tested against both PAGE_V1 and PAGE_V2, this heuristics seems working OK, an ideal solution is to use PageEncodingStats, this is a heuristics workaround. I put todo here as a next step